packages/react/src/components/pickers/BasePicker.tsx

```diff
@@ -597,6 +598,10 @@ export class BasePicker<T, P extends IBasePickerProps<T>>
     if (this.props.inputProps !== undefined && this.props.inputProps.onClick !== undefined) {
       this.props.inputProps.onClick(ev);
     }
+    // Reset focus to last item if the input is removed
+    if (!this.canAddItems()) {
+      this.resetFocus(this.state.items.length - 1);
+    }
```

Comment:
I think you should put this into a new `onWrapperClick` function instead of attaching it here.